external-crates/move/move-compiler/src/cfgir/cfg.rs

```diff
@@ -283,14 +283,25 @@ const DEAD_ERR_EXP: &str = "Invalid use of a divergent expression. The code foll
                             evaluation of this expression will be dead and should be removed.";
 
 fn dead_code_error(diags: &mut Diagnostics, block: &BasicBlock) {
-    let first_command = block.front().unwrap();
-    match unreachable_loc(first_command) {
-        Some(loc) => diags.add(diag!(UnusedItem::DeadCode, (loc, DEAD_ERR_EXP))),
-        None if is_implicit_control_flow(block) => (),
-        None => diags.add(diag!(
-            UnusedItem::DeadCode,
-            (first_command.loc, DEAD_ERR_CMD)
-        )),
+    match build_dead_code_error(block) {
+        Some(diag) => diags.add(diag),
+        None => (),
+    }
+}
+
+pub fn build_dead_code_error(block: &BasicBlock) -> Option<Diagnostic> {
+    if block.len() < 1 {
+        None
+    } else {
```

Comment:
We need empty blocks because sometimes blocks are just... empty. This is an artifact of how unreachable code is handled in lowering to HLIR: we only insert returns when the end is reachable, so sometimes we'll end up with code like:

```
fun foo(...) {
  loop {}
}
```

The HLIR lowering declares this unreachable and doesn't synthesize a return, so when we go to build the blocks "backwards", we have no base block for the exit label on the loop. We need to allow an empty block there so that the exit label for the loop is a valid block label.

We can get rid of this and write the block generation nicer if we could guarantee all functions end with _some_ sort of terminal command, even if that command is "I am unreachable." If we had that, we could also generate the block graph in a slightly nicer way. I figured I'd leave it for the future, though, predicated on some discussion around pushing assignments into blocks and whatnot.